app/models/error_detail.rb

```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+class ErrorDetail < ApplicationRecord
+  include Discard::Model
+  self.discard_column = :deleted_at
+
+  belongs_to :owner, polymorphic: true
+  belongs_to :organization
+
+  ERROR_CODES = %w[not_provided tax_error]
+  enum error_code: ERROR_CODES.zip(ERROR_CODES).to_h
```

Comment:
Just realised that string DB field is used to be an enum. Natively in Rails enums are stored as integers and that would allow us to avoid such syntax. Is there any additional reason for using it this way? If not, I would prefer using enums as we use everywhere else across the app:)

Reply:
with enums - we don't know how much this codes will grow and how often we'll need to add them, because every time we're adding a new error type, we'll need to update the enum table.
If it's okay I'm happy to update 😁

Reply:
If I haven't missed something, In both cases, we will only have to update `ERROR_CODES` constant. It is just matter how we store it and how enum definition looks like:)